### PR TITLE
Add MALA (Material Learning Algorithms)

### PR DESCRIPTION
**Pull request by franzpoeschel**

### CODE REVIEW

**Review by ax3l — APPROVED** [review]
Excellent, thank you!

Discussed offline: Waiting for MALA devs to confirm/review.

**Comment by franzpoeschel** [comment]
Pinging @acangi and @RandomDefaultUser for your input on this

**Comment by RandomDefaultUser** [comment]
Thanks for including MALA! This looks great to me with one exception: In the first line, ``(materials learning algorithms)`` is given at the end of the domain description. "materials learning algorithms" is not a domain-specific term, but simply the written out form of MALA. I would remove it from the first line and either not mention it or in the header line.

Edit: Added a suggestion to this effect, didn't realize I had access rights for that when writing this comment.

File: README.md [hunk 1]
```diff
@@ -140,6 +140,12 @@ the work provided in those for our format handling.
   - original code authors: James Amundson, Steve Goldhaber, Paul Lebrun, Qiming Lu, Alexandru Macridin, Leo Michelotti (CHEF libraries), Chong Shik Park, Panagiotis Spentzouris and Eric Stern
   - status: openPMD I/O is implemented for bunch I/O and statistics. Internal checkpoints still use custom HDF5 formats, but these would likely be moved to openPMD in the future.
 
+- [MALA](https://mala-project.github.io/mala/) (CASUS, SNL)
+  - domain: ML models that replace density functional theory (DFT) calculations (materials learning algorithms)
```
Comment (RandomDefaultUser) [inline README.md]:
```suggestion
- [MALA (Materials Learning Algorithms)](https://mala-project.github.io/mala/) (CASUS, SNL)
  - domain: ML models that replace density functional theory (DFT) calculations
```

**Comment by franzpoeschel** [comment]
Thank you Lenz, I've added your suggestion. I think it's alright to have the full name somewhere, given that the name is somewhat informative about what the software does ;)

**Pull request merged**